UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -280,7 +285,23 @@ private async Task CheckProductOrCancellationData(SearchBatchResponse internalSe
                 internalSearchBatchResponse.Entries.Add(item);
                 productList.Add(compareProducts);
                 if (CommonHelper.IsPeriodicOutputService)
```

Comment:
This logic from 288-307 should be split out into its own method to reduce the complexity in  `CheckProductOrCancellationData`